### PR TITLE
docs: Add docs for Argo Workflow controller-instanceid label (#290)

### LINKED ISSUE #290: Add documentation for the 'workflows.argoproj.io/controller-instanceid' label requirement in Argo Workflows
## Is this a BUG REPORT or FEATURE REQUEST?
 
 FEATURE REQUEST

**What happened**:

While creating and applying an Argo Workflow without **ChaosCenter**, the Workflow controller didn't pick up the Workflow file. After investigating and receiving feedback from the LitmusChaos community, it was discovered that the `workflows.argoproj.io/controller-instanceid` label was missing in the `metadata.labels` section of the Workflow YAML. This requirement wasn't documented, causing confusion.

**What you expected to happen**:

I expected the Workflow controller to pick up and manage the Workflow without any issues when applied without **ChaosCenter**. I also expected the documentation to include all necessary configuration steps, including the need for the `workflows.argoproj.io/controller-instanceid` label.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a Workflow YAML file without including the `workflows.argoproj.io/controller-instanceid` label under `metadata.labels`.
2. Apply the Workflow using `kubectl apply -f workflow.yaml`.
3. Observe that the Workflow controller does not pick up the Workflow.

**Anything else we need to know?**:

Including the `workflows.argoproj.io/controller-instanceid` label resolved the issue. The value for this label can be found in the `workflow-controller-configmap` under the instanceID key.

### PR DESCRIPTION
**Pull request by thiha-min-thant**
**What this PR does / why we need it**:

- Added a section to explain the requirement of the `workflows.argoproj.io/controller-instanceid` label in the Workflow YAML.
- Included an example showing where to add the label.
- Explained how to retrieve the instanceID value from the workflow-controller-configmap.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #290 

**Special notes for your reviewer**:

**Checklist:**
-   [x] Fixes #290 
-   [x] Signed the commit for DCO to be passed
-   [ ] Labelled this PR & related issue with `documentation` tag

### CODE REVIEW

**Review by bjoky — APPROVED** [review]

File: website/docs/faq.md [hunk 1]
```diff
@@ -70,6 +70,30 @@ To configure a git repo the user must provide the Git URL of the repository and
 
 Once GitOps is enabled, any new chaos experiments created will be stored in the configured repo in the path `litmus/<project-id>/<chaos-experiment-name>.yaml`.
 
+### Why is my Argo Workflow not being picked up by the Workflow controller?
```
Comment (neelanjan00) [inline website/docs/faq.md]:
```suggestion
### Why is my Argo Workflow not being picked up by the Workflow controller upon applying it manually?
```

Comment (neelanjan00) [reply website/docs/faq.md]: Same change in the versioned doc as well, please.

File: website/docs/faq.md [hunk 2]
```diff
@@ -70,6 +70,30 @@ To configure a git repo the user must provide the Git URL of the repository and
...
+
+If your Argo Workflow is not being picked up by the Workflow controller after applying it, you may be missing the `workflows.argoproj.io/controller-instanceid` label in your Workflow YAML file. This label is required for the Workflow controller to manage the Workflow properly.
```
Comment (neelanjan00) [inline website/docs/faq.md]:
```suggestion
If your Argo Workflow is not being picked up by the Workflow controller after applying it, you may be missing the `workflows.argoproj.io/controller-instanceid` label in your Workflow manifest. This label is required for the Workflow controller to reconcile the Workflow upon its creation.
```

File: website/docs/faq.md [hunk 3]
```diff
@@ -70,6 +70,30 @@ To configure a git repo the user must provide the Git URL of the repository and
...
+
+To fix this, ensure the following label is added under metadata.labels in your Workflow YAML:
```
Comment (neelanjan00) [inline website/docs/faq.md]:
```suggestion
To fix this, ensure the following label is added under `metadata.labels` field in the Workflow manifest:
```

File: website/docs/faq.md [hunk 4]
```diff
@@ -70,6 +70,30 @@ To configure a git repo the user must provide the Git URL of the repository and
...
+    workflows.argoproj.io/controller-instanceid: <instanceID>
+```
+
+The `instanceID` value can be found in the `workflow-controller-configmap`. Here’s how to retrieve it:
+
+```shell
+kubectl get configmap workflow-controller-configmap -n <namespace> -o yaml
+```
+
+Look for the `instanceID` key in the configmap and use that value in your Workflow YAML.
```
Comment (neelanjan00) [inline website/docs/faq.md]:
```suggestion
Look for the `instanceID` key in the configmap and use that value in your Workflow manifest.
```

File: website/docs/faq.md [hunk 5]
```diff
@@ -70,6 +70,30 @@ To configure a git repo the user must provide the Git URL of the repository and
...
+
+Without this label, the Workflow controller will not recognize or manage the Workflow.
```
Comment (neelanjan00) [inline website/docs/faq.md]:
```suggestion
Without this label, the Workflow controller will not be able to reconcile the Workflow.
```

File: website/docs/user-guides/construct-experiment.md [hunk 6]
```diff
@@ -65,19 +65,24 @@ Here, the template with the name `custom-chaos` will be executed first.
...
+           data: >
+             apiVersion: litmuschaos.io/v1alpha1
+
+             description:
+               message: |...
 ```
 
+### Ensuring Your Workflow is Recognized by the Argo Workflow Controller
+
+When applying a Workflow manually without ChaosCenter, it's crucial to include the `workflows.argoproj.io/controller-instanceid` label in your YAML file. This label helps the controller identify and manage your Workflow correctly.
```
Comment (neelanjan00) [inline website/docs/user-guides/construct-experiment.md]:
```suggestion
When applying a Workflow manually without ChaosCenter, it's crucial to include the `workflows.argoproj.io/controller-instanceid` label in the manifest. This label helps Argo Workflow controller identify and reconcile the Workflow upon its creation.
```

**Review by neelanjan00 — APPROVED** [review]
LGTM, minor nits. Please make the changes in both master and versioned files.

**Comment by thiha-min-thant** [comment]
Thanks for the feedback, @neelanjan00 !  I've made the necessary updates based on your suggestions.

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**